src/include/txn_inline.h

```diff
@@ -1161,10 +1152,8 @@ __wt_txn_read(
      */
     if (prepare_upd != NULL) {
         WT_ASSERT(session, F_ISSET(prepare_upd, WT_UPDATE_PREPARE_RESTORED_FROM_DS));
-        if (retry &&
```

Comment:
Sorry I am not familiar with this code, why can we remove this `retry` variable now?

Reply:
This retry was put there to ensure we can only retry once. Now with the change, we may retry because of different reasons for more than once. We should remove this.